components/google-cloud/google_cloud_pipeline_components/experimental/custom_job/custom_job.py

```diff
@@ -118,13 +119,29 @@ def _is_output_parameter(output_key: str) -> bool:
                 if 'args' in container_spec:
                     dsl_utils.resolve_cmd_lines(container_spec['args'],
                                                 _is_output_parameter)
+                    # Temporarily remove {{{{$}}}} executor_input arg as it is not supported by the backend.
+                    logging.info(
+                        "Removing executor input, as it is currently not supported by the backend."
+                    )
+                    container_spec['args'] = [
+                        arg for arg in container_spec['args']
+                        if arg not in ('{{{{$}}}}', '--executor_input')
```

Comment:
This is due to a nuance with python string formatter and place holders each "{{" translates to a "{" as a string literal. so to get "{{$}}" we need to have "{{{{$}}}}" therefore the placeholder pre IR is actually "{{{{$}}}}" instead of "{{$}}".

Ref: https://docs.python.org/3/library/string.html#formatstrings

In our code the conversion happens at : https://github.com/kubeflow/pipelines/blob/2f19a26ffd205c7a3384c4c250f5dbb5150a7c4d/sdk/python/kfp/dsl/dsl_utils.py#L101-L104